### ISSUE TITLE
Planet Surface with negative soil

### ISSUE THREAD

**Issue by morden96** [Touhma/DSP_Plugins#135]
On a Ashen Gelisol planet (100% buildable area), the lower lying areas of the planet consume soil when building. As a result, by mech has a negative soil balance after building alot and cannot place foundations. Even on high-soil areas, foundations fail to place because my soil pile will be negative after placement (because it's already so negative).

I have several other mods, but none of them affect soil. The only mod that affects planet generation is the SquarePatchMod.

![image](https://user-images.githubusercontent.com/6688940/129496909-91b9abd9-fe4a-4b9a-a12b-eb52f32446fd.png)

**Comment by innominata** [comment]
[GalacticScaleTestBackupSavesFirst.zip](https://github.com/Touhma/DSP_Plugins/files/6989922/GalacticScaleTestBackupSavesFirst.zip)
This might be related to another bug with flattening terrain on huge planets, it was fixed a while ago but a recent update has broken it. I've fixed the fix in the attached DLL, but I don't know if it will do strange things to existing terrain levels, so backup your saves first

**Comment by morden96** [comment]
I think I'm installing it wrong. I get this error when I put it in
BepInEx\plugins\Galactic_Scale-GalacticScale:
[image: image.png]

**Comment by morden96** [comment]
Scratch that. This error is produced by v2.0.862, even without the .dll you
sent me. Error does not appear in v2.0.860. Reinstalled and confirmed.

**Comment by innominata** [comment]
Doesn't look like the picture you send worked

However, the original issue isn't GS2 related, it happens in vanilla also
![image](https://user-images.githubusercontent.com/4099099/129556496-4e39302e-a466-4b93-b4ca-057ee9fe9627.png)

**Comment by innominata** [comment]
If you can post the error in #support-bugs on our discord server, it would really help
https://discord.gg/XBUafuNzbA

**Comment by morden96** [comment]
Sorry, my discord is crashing.... (?)

**Comment by innominata** [comment]
> This error is also produced by v2.0.862, even without the .dll you sent me. Error does not appear in v2.0.860. Reinstalled and confirmed.

What error?

**Comment by morden96** [comment]
The error I posted yesterday when I tried the dll

**Comment by innominata** [comment]
Yeah, you didn't post anything. At least nothing that shows up for me
![image](https://user-images.githubusercontent.com/4099099/130008149-2073931f-67cd-43d5-a4c7-532fa7b6649f.png)
